02-documentation/developer-guides/building-a-nextjs-blog-with-squidex.md

```diff
@@ -0,0 +1,588 @@
+
+```bash
+yarn add next-mdx-remote
+```
+
+Restart the development server after installation:
+
+```
+yarn dev
+```
+
+Import `serialize` and `MDXRemote` from `next-mdx-remote` at the top of `[slug].js`:
+
+```jsx
+import { serialize } from "next-mdx-remote/serialize";
+import { MDXRemote } from "next-mdx-remote";
+```
+
+Import `fetchAPI` at the top of `[slug].js` as you will use it to fetch the contents your blog posts from Squidex:
+
+```js
+import fetchAPI from "../lib/squidex";
+```
+
+Export a `getStaticProps` function from `[slug].js`:
+
+```js
+export async function getStaticProps({ params }) {
+  const data = await fetchAPI(
+    `
+  query QueryPosts($query: String!) {
+    queryPostsContents(filter: $query) {
+      id
+      flatData {
+        title
+        content
+      }
+    }
+  }
+  `,
+    {
+      variables: {
+        query: `data/Slug/iv eq '${params.slug}'`,
+      },
+    }
+  );
+
+  if (data.queryPostsContents.length === 0) {
+    return {
+      notFound: true,
+    };
+  }
+  const post = data.queryPostsContents[0].flatData;
+
+  const mdxSource = await serialize(post.content);
+
+  return {
+    props: {
+      post: post,
+      source: mdxSource,
+    },
+  };
+}
+```
+
+This returns data for the blog post associated with a particular slug or returns a 404 error if no such post exists.
+
+At this point, you may receive an error because when you use `getStaticPaths` with dynamic paths in Next.js, you are required to also export a `getStaticPaths` function that tells Next.js [what pages to generate at build time](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation).
+
+Export a `getStaticPaths` function in `squidex-blog/pages/index.js`:
+
+```js
+export async function getStaticPaths() {
+  const data = await fetchAPI(`
+  {
+    queryPostsContents {
+      id,
+      flatData {
+        slug
+      }
+    }
+  }
+  `);
+
+  return {
+    paths: data.queryPostsContents.map((post) => {
+      return {
+        params: {
+          slug: post.flatData.slug,
+        },
+      };
+    }),
+    fallback: false,
+  };
+}
+```
+
+When you save the file, refresh the page to fetch data from Squidex. Right now, again, although the data is available to the page, you cannot see it since we have not rendered the blog post. You will now render the loaded blog post on the page.
+
+Modify the `BlogPost` function in `[slug].js`:
+
+```jsx
+export default function BlogPost({ post, source }) {
+  return (
+    <div>
+      <h1 className="blog-post-title">{post.title}</h1>
+      <MDXRemote {...source} />
+    </div>
+  );
+}
+```
+
+Once you save the file, the title of the blog post and the body of the post will show on the page. If you did not refresh the page after adding `getStaticProps`, you may see an error. Refresh the page to fetch data from your Squidex application.
```

Comment:
should be "will be rendered on the page" (I think). 

And why do I need to refresh all the time?